workflow/controller/controller.go

```diff
@@ -763,10 +841,20 @@ func (wfc *WorkflowController) getMetricsServerConfig() (metrics.ServerConfig, m
 		Port:         port,
 		IgnoreErrors: wfc.Config.TelemetryConfig.IgnoreErrors,
 	}
-
 	return metricsConfig, telemetryConfig
 }
 
+func (wfc *WorkflowController) cleanupWorkflowDeletion(obj interface{}) {
```

Comment:
I was also confused on what this was supposed to do because the name is not reflective of what it was doing. I suggest: releaseAllWorkflowLocks()